matrix/matrix.py

```diff
@@ -0,0 +1,161 @@
+"""Matrix data type class."""
+
+import operator
+
+
+class Matrix:
+    """An immutable object representing a matrix data-type.
+
+    Attributes:
+        _matrix (tuple): contains the data about the matrix. Each line of
+            the matrix is represented by a tuple.
+    Note:
+        This implementation only supports squared matrices (n * n).
+    """
+
+    def __init__(self, lines):
+        self._validate(lines)
+        self._matrix = lines
+
+    def _validate(self, data: tuple):
+        """Check if the data given is valid.
+
+        Note:
+            Valid data is defined if it contains tuples of the same length
+            (Matrix only supports n * n), items can't be iterable.
+        """
+        if any(hasattr(item, '__len__') for line in data for item in line):
```

Comment:
Also not Pythonic
Python will raise an exception if the object doesn't have `__len__`, so you don't need to implement anything by yourself

Reply:
So ignore the case of an iterable as a marix item?

Reply:
You anyway iterate over it, so if it's not iterable some exception will be raised

Reply:
yes